op-bindings/etherscan/client.go

```diff
@@ -0,0 +1,211 @@
+package etherscan
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/op-service/retry"
+)
+
+type client struct {
+	baseUrlEth string
+	baseUrlOp  string
+}
+
+type apiResponse struct {
+	Status  string      `json:"status"`
+	Message string      `json:"message"`
+	Result  interface{} `json:"result"`
+}
+
+type rpcResponse struct {
+	JsonRpc string      `json:"jsonrpc"`
+	Id      int         `json:"id"`
+	Result  interface{} `json:"result"`
+}
+
+type TxInfo struct {
+	To    string `json:"to"`
+	Input string `json:"input"`
+}
+
+const apiMaxRetries = 3
+const apiRetryDelay = time.Duration(2) * time.Second
+
+func NewClient(apiKeyEth, apiKeyOp string) *client {
```

Comment:
Why must this operate with both eth and op? Its kinda weird, why not have 2 clients that both take a url and an api key?

Reply:
This makes it generic and reusable. Could even wrap the generic `NewClient` with a
```
func NewEthereumClient(apiKey string) *client {
    return NewClient(EthereumEtherscanURL, apiKey)
}
```

Reply:
Yea, the implementation was specific to what BindGen needed, but there's no reason for it to be this way - will update, thank you!